cmd/agent/app/reporter/grpc/collector_proxy_test.go

```diff
@@ -80,3 +81,15 @@ func TestMultipleCollectors(t *testing.T) {
 	assert.Equal(t, true, bothServers)
 	require.Nil(t, proxy.Close())
 }
+
+func initializeGRPCTestServer(t *testing.T, beforeServe func(server *grpc.Server)) (*grpc.Server, net.Addr) {
+	server := grpc.NewServer()
+	lis, err := net.Listen("tcp", "localhost:0")
+	require.NoError(t, err)
+	beforeServe(server)
+	go func() {
+		err := server.Serve(lis)
```

Comment:
You can just use `server.Serve(list)` directly as a parameter on the next line: `require.NoError(t, server.Serve(list))`